tests/components/influxdb/test_init.py

```diff
@@ -11,748 +11,1116 @@
     STATE_STANDBY,
     UNIT_PERCENTAGE,
 )
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import get_test_home_assistant
+from tests.async_mock import MagicMock, Mock, call, patch
 
+BASE_V1_CONFIG = {}
+BASE_V2_CONFIG = {
+    "api_version": influxdb.API_VERSION_2,
+    "organization": "org",
+    "token": "token",
+}
 
-@mock.patch("homeassistant.components.influxdb.InfluxDBClient")
-@mock.patch(
-    "homeassistant.components.influxdb.InfluxThread.batch_timeout",
-    mock.Mock(return_value=0),
-)
-class TestInfluxDB(unittest.TestCase):
-    """Test the InfluxDB component."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.handler_method = None
-        self.hass.bus.listen = mock.Mock()
-
-    def tearDown(self):
-        """Clear data."""
-        self.hass.stop()
-
-    def test_setup_config_full(self, mock_client):
-        """Test the setup with full configuration."""
-        config = {
-            "influxdb": {
-                "host": "host",
-                "port": 123,
-                "database": "db",
+
+@pytest.fixture(autouse=True)
+def mock_batch_timeout(hass, monkeypatch):
+    """Mock the event bus listener and the batch timeout for tests."""
+    hass.bus.listen = MagicMock()
+    monkeypatch.setattr(
+        "homeassistant.components.influxdb.InfluxThread.batch_timeout",
+        Mock(return_value=0),
+    )
+
+
+@pytest.fixture
+def mock_client(request):
```

Comment:
We don't need to disable pylint redefined-outer-name
 like this:
```suggestion
@pytest.fixture(name="mock_client")
def mock_client_fixture(request):
```

Reply:
Oh nice! Yea that was bugging me, I didn't realize you could put the fixture name in the annotation.